classwork/classwork3.2/src/TriangleSuccess.cpp

```diff
@@ -0,0 +1,14 @@
+#include "Triangle.h"
+#include <cmath>
+
+double Triangle::area() const {
+    if (a <= 0 || b <= 0 || c <= 0) {
+        return -1; // Indicate invalid triangle
+    }
+    if (a + b <= c || a + c <= b || b + c <= a) {
+        return -1; // Indicate invalid triangle
```

Comment:
optional is a better way to represent errors of such kind. Therefore std::optional<double> is the real return type not double.
Why it is important? If you are using this convention to use -1(of course that should be some named constexpr constant not hardcoded value in multiple places) as invalid value then every user should not about it. At the same anyone who sees optional<double> will instantly understand that this function may or may not return double